### PR TITLE
refactor: GießDenKiez -> NextJS

### PR DESCRIPTION
**Pull request by vogelino**
This PR refactors GDK to use NextJS instead of Create React App.

- [x] Update of env vars with NEXT_PUBLIC_ prefix
- [x] React router is gotten rid of and we now use file based routing provided by NextJS
- [x] Jest and Eslint work with NextJs setup
- [x] GitHub Actions are adapted to work with the new setup
- [x] Webpack is removed
- [x] Node is updated to version 18
- [x] Babel is removed
- [x] We are using the next/link component
- [x] We are using the next/image component
- [x] Styled components works server side
- [x] Storybook is updated to work with the new setup
- [x] We use the layout feature to share the map between routes

### CODE REVIEW

**Comment by vogelino** [comment]
> * @SocketSecurity ignore next@12.3.2

Please my dear robot, @SocketSecurity ignore next@12.3.2

File: create-env.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
-/* eslint-disable @typescript-eslint/no-var-requires */
 const fs = require('fs');
 const envVars = `
 # these are only generated on netlify deploy
```
Comment (dnsos) [inline create-env.js:3]: If we move to Vercel with this PR, we should remove this file. And thinking about it, probably everything that is Netlify-related...

File: docs/users-stats/global.d.ts [hunk 2]
```diff
@@ -1512,13 +1512,13 @@ declare namespace Deno {
   export function truncate(name: string, len?: number): Promise<void>;
 
   export interface NetAddr {
-    transport: "tcp" | "udp";
+    transport: 'tcp' | 'udp';
```
Comment (dnsos) [inline docs/users-stats/global.d.ts:1515]: I know it's just formatting, but do you know  what this file is for? I hadn't encountered it before

**Comment by vogelino** [comment]
@SocketSecurity ignore next@12.3.2

**Review by dnsos — APPROVED** [review]
Incredible work @vogelino! It's such a good step to move this into the safe haven of a good framework. This will make further refactoring much more approachable.

I reviewed the changes as thorough as possible and tested the UI manually. Everything works smooth and as expected. Great job!

In general, I think we can move on with the plan to migrate also the hosting with this change, so let's talk about the next steps. On that, I actually believe that we have to fine-tune the environment variables a bit, because currently the deploy preview shows only a small portion of trees rendered. Perhaps there is a mixup with the dev and prod tilesets?

**Comment by vogelino** [comment]
> Incredible work @vogelino!

Thx! Appreciate it!

> I actually believe that we have to fine-tune the environment variables a bit, because currently the deploy preview shows only a small portion of trees rendered

Good point. I just noticed that too. I'll have a look!

File: create-env.js (hunk 1 above)
Comment (vogelino) [reply create-env.js:3]: Agreed! I kept it here as it wasn't yet clear if we would deploy that on Netlify or not. I'll remove it!

File: docs/users-stats/global.d.ts (hunk 2 above)
Comment (vogelino) [reply docs/users-stats/global.d.ts:1515]: TBH no idea 🤷‍♂️ 
@ff6347 What about you?

**Review by ff6347 — APPROVED** [review]

**Pull request merged**